examples/email-mcp/test/inbound-route.test.ts

```diff
@@ -0,0 +1,55 @@
+/// <reference types="@cloudflare/workers-types" />
+import { describe, it, expect, beforeAll } from "vitest";
+import { EmailRepository } from "../src/repository";
+
+// Vitest Workers pool injects global `SELF` fetch (the worker) and `env`.
+declare const env: Env;
+// `SELF` is the worker module’s default export object that handles fetch/email in the pool.
+declare const SELF: ExportedHandler<Env>;
```

Comment:
Look into the examples/crud-mcp/tests and create a tool based test to test all of the tools and prompts (AI should do well if you reference that project to recreate this).

Regarding the HTTP endpoints, looks fine. but you need to cover error cases IMO